### PR TITLE
write request keep compatible with before

### PR DESCRIPTION
**Pull request by song-zhang**

### CODE REVIEW

**Comment by chrislusf** [comment]
seems the fsync is still slow?

```
~$ weed benchmark -fsync
This is SeaweedFS version 30GB 1.77 darwin amd64

------------ Writing Benchmark ----------
Completed 167 of 1048576 requests, 0.0% 167.0/s 0.2MB/s
Completed 347 of 1048576 requests, 0.0% 179.6/s 0.2MB/s
Completed 534 of 1048576 requests, 0.1% 187.4/s 0.2MB/s
Completed 743 of 1048576 requests, 0.1% 209.0/s 0.2MB/s
Completed 948 of 1048576 requests, 0.1% 205.0/s 0.2MB/s
Completed 1128 of 1048576 requests, 0.1% 180.0/s 0.2MB/s
Completed 1311 of 1048576 requests, 0.1% 183.0/s 0.2MB/s
Completed 1495 of 1048576 requests, 0.1% 184.0/s 0.2MB/s
^C~$ weed benchmark -fsync=false
This is SeaweedFS version 30GB 1.77 darwin amd64

------------ Writing Benchmark ----------
Completed 21186 of 1048576 requests, 2.0% 21170.5/s 21.3MB/s
Completed 40508 of 1048576 requests, 3.9% 19333.0/s 19.5MB/s
Completed 62546 of 1048576 requests, 6.0% 22032.9/s 22.2MB/s
Completed 84566 of 1048576 requests, 8.1% 22026.7/s 22.2MB/s
Completed 105585 of 1048576 requests, 10.1% 21019.0/s 21.2MB/s
Completed 127265 of 1048576 requests, 12.1% 21681.9/s 21.8MB/s
Completed 148048 of 1048576 requests, 14.1% 20781.5/s 20.9MB/s
^C~$
```

**Comment by song-zhang** [comment]
a same test in my server:

before:
./weed benchmark -fsync

Concurrency Level:      16
Time taken for tests:   788.631 seconds
Complete requests:      1048576
Failed requests:        0
Total transferred:      1106777287 bytes
Requests per second:    1329.62 [#/sec]
Transfer rate:          1370.52 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        0.8      12.0       275.7      10.0

Percentage of the requests served within a certain time (ms)
   50%      8.5 ms
   66%     13.0 ms
   75%     16.5 ms
   80%     18.7 ms
   90%     25.4 ms
   95%     31.5 ms
   98%     39.6 ms
   99%     46.1 ms
  100%    275.7 ms

now:
./weed benchmark -fsync

Concurrency Level:      16
Time taken for tests:   512.727 seconds
Complete requests:      1048576
Failed requests:        0
Total transferred:      1106768387 bytes
Requests per second:    2045.10 [#/sec]
Transfer rate:          2108.00 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        0.8      7.8       241.5      5.6

Percentage of the requests served within a certain time (ms)
   50%      6.2 ms
   66%      7.6 ms
   75%      9.0 ms
   80%     10.0 ms
   90%     13.8 ms
   95%     18.7 ms
   98%     24.3 ms
   99%     29.0 ms
  100%    241.5 ms

50% performance increase

**Comment by song-zhang** [comment]
decrease volume number in HDD disk or increase concurrency can dramatically improve performance:

1.with only on volume
curl "http://localhost:9333/vol/grow?collection=benchmark1&count=1"
./weed benchmark -fsync -collection=benchmark1

Concurrency Level:      16
Time taken for tests:   250.770 seconds
Complete requests:      1048576
Failed requests:        0
Total transferred:      1106788965 bytes
Requests per second:    4181.43 [#/sec]
Transfer rate:          4310.12 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        0.9      3.8       107.2      4.8

Percentage of the requests served within a certain time (ms)
   50%      2.1 ms
   66%      2.7 ms
   75%      3.4 ms
   80%      4.5 ms
   90%      7.9 ms
   95%     12.8 ms
   98%     17.1 ms
   99%     24.1 ms
  100%    107.2 ms

2.use more concurrency:
./weed benchmark -fsync -c=1024

Concurrency Level:      1024
Time taken for tests:   50.973 seconds
Complete requests:      1048576
Failed requests:        0
Total transferred:      1106758999 bytes
Requests per second:    20571.20 [#/sec]
Transfer rate:          21203.76 [Kbytes/sec]

Connection Times (ms)
              min      avg        max      std
Total:        0.9      49.5       304.1      35.0

Percentage of the requests served within a certain time (ms)
   50%     36.9 ms
   66%     48.8 ms
   75%     61.1 ms
   80%     72.3 ms
   90%    103.7 ms
   95%    125.1 ms
   98%    149.2 ms
   99%    165.7 ms
  100%    304.1 ms

**Comment by chrislusf** [comment]
can you use the latest master branch? My number is drastically different.

**Comment by chrislusf** [comment]
@song-zhang could you help to add better checking for the volume file size?

This could be related to https://github.com/chrislusf/seaweedfs/issues/1349

**Comment by chrislusf** [comment]
Need to revert this. See #1349  for details.

File: weed/storage/volume_read_write.go [hunk 1]
```diff
@@ -129,14 +127,18 @@ func (v *Volume) writeNeedle2(n *needle.Needle, fsync bool) (offset uint64, size
 		n.Ttl = v.Ttl
 	}
 
-	asyncRequest := needle.NewAsyncRequest(n, true, fsync)
-	// using len(n.Data) here instead of n.Size before n.Size is populated in n.Append()
-	asyncRequest.ActualSize = needle.GetActualSize(uint32(len(n.Data)), v.Version())
+	if !fsync {
```
Comment (chrislusf) [inline weed/storage/volume_read_write.go:130]: This is wrong?

**Pull request merged**